src/s2/s2region_term_indexer.h

```diff
@@ -289,6 +303,21 @@ class S2RegionTermIndexer {
   std::vector<std::string> GetQueryTermsForCanonicalCovering(
       const S2CellUnion& covering, absl::string_view prefix);
 
+  // Same as above but allows to reuse same buffer for different points or use
```

Comment:
I think its fine to have these since the old ones call the new variants.  Since the terms vector isn't optional let's just make it a non-const reference though.

Reply:
Ah yes if the performance gain isn't worth it then we should probably just elide.

Reply:
> Are they cleared? appended to?

appended to, I think function is pretty small to see it from code.

> The formatting is also inconsistent (const T& vs const T &) with a few lines below. A lot of time is spent dealing with this for epsilon difference.

Sorry you don't have clang-format in repo, so my ide reformat it randomly, I tried to make it like was in other function, maybe in some places I missed it

> This seems like a small inefficiency we should forget about 97% of the time. @MBkkt even said the few microseconds it saved were irrelevant to him.

> This seems like a small inefficiency we should forget about 97% of the time. @MBkkt even said the few microseconds it saved were irrelevant to him.

https://github.com/google/s2geometry/pull/303#discussion_r1069346176

I already wrote, now it's ok for me don't have such api for query, but for indexing, calling other api make my code faster (a little but measurable), so I want other api.



### Offtop:

Btw today we discussed how speedup geo terms search: now we need to make disjunction of all query terms (it's from s2 documentation and it works now for us).
But it's possible to make conjuntion of disjunction which is a lot of better for inverted index (don't need to read a lot of postings lists)

Like query covering:
```
cellid-0: min_level 
cellid-1: min_level + 1
cellid-2: min_level + 1
```
Like query terms:
```
cellid-0-term-min-level
cellid-1-term-min-level
cellid-1-term-min-level+1
// cellid-2-term-min-level, for an example same as cellid-1-term-min-level, so deduplicated
cellid-2-term-min-level+1
```
So we can make query like:
```
(cellid-0-term-min-level || cellid-1-term-min-level) && (cellid-1-term-min-level+1 || cellid-2-term-min-level+1)
```
instead of 
```
cellid-0-term-min-level || cellid-1-term-min-level || cellid-1-term-min-level+1 || cellid-2-term-min-level+1
```

It has few other tricks, like we should store in index for pair: term, document in posting list: is it last or not

But in general it's a lot of better.

Why do I write about this?

Because for use such api we need to know term level.
It can be computed from terms, but it's not really nice, because we already know they from covering.

So will be nice to have overload in future like:

std::vector<std::pair<int/*level*/,std::string/*term*/>>

Reply:
> I think its fine to have these since the old ones call the new variants. Since the terms vector isn't optional let's just make it a non-const reference though.

@smcallis 

reference looks better for me, but in all other places in s2 mutable reference arguments passed like pointer